src/util/include/util/Color.h

```diff
@@ -11,149 +11,160 @@
+        return uint32_t(lhs) == uint32_t(rhs);
     }
-    template <typename T, std::enable_if_t<std::is_unsigned_v<T> && !std::is_same_v<T, uint32_t>, int> = 0>
-    constexpr explicit operator T() const {
-        return val;
+    constexpr friend bool operator!=(LegacyARGBU8Color const& lhs, LegacyARGBU8Color const& rhs) {
+        return uint32_t(lhs) != uint32_t(rhs);
     }
-    template <typename T, std::enable_if_t<std::is_signed_v<T>, int> = 0>
-    constexpr explicit operator T() const {
-        return val;
+    constexpr friend bool operator<(LegacyARGBU8Color const& lhs, LegacyARGBU8Color const& rhs) {
+        return uint32_t(lhs) < uint32_t(rhs);
     }
 
-    constexpr auto operator=(Color const& rhs) -> Color& = default;
-    constexpr auto operator=(uint32_t rhs) -> Color& { return *this = Color(rhs); }
-
-    constexpr friend auto operator&(Color lhs, uint32_t rhs) -> uint32_t { return lhs.val & rhs; }
-    constexpr friend auto operator|(Color lhs, uint32_t rhs) -> uint32_t { return lhs.val | rhs; }
-    constexpr friend auto operator^(Color lhs, uint32_t rhs) -> uint32_t { return lhs.val ^ rhs; }
-    constexpr friend auto operator&(uint32_t lhs, Color rhs) -> uint32_t { return lhs & rhs.val; }
-    constexpr friend auto operator|(uint32_t lhs, Color rhs) -> uint32_t { return lhs | rhs.val; }
-    constexpr friend auto operator^(uint32_t lhs, Color rhs) -> uint32_t { return lhs ^ rhs.val; }
-    constexpr friend auto operator>>(Color lhs, uint32_t rhs) -> uint32_t { return lhs.val << rhs; }
-    constexpr friend auto operator<<(Color lhs, uint32_t rhs) -> uint32_t { return lhs.val >> rhs; }
+    constexpr friend bool operator==(LegacyARGBU8Color const& lhs, uint32_t rhs) { return uint32_t(lhs) == rhs; }
+    constexpr friend bool operator!=(LegacyARGBU8Color const& lhs, uint32_t rhs) { return uint32_t(lhs) != rhs; }
+    constexpr friend bool operator<(LegacyARGBU8Color const& lhs, uint32_t rhs) { return uint32_t(lhs) < rhs; }
 
-    constexpr friend bool operator==(Color lhs, Color rhs) { return lhs.val == rhs.val; }
-    constexpr friend bool operator!=(Color lhs, Color rhs) { return lhs.val != rhs.val; }
-    constexpr friend bool operator<(Color lhs, Color rhs) { return lhs.val < rhs.val; }
-    constexpr friend bool operator<=(Color lhs, Color rhs) { return lhs.val <= rhs.val; }
-    constexpr friend bool operator>(Color lhs, Color rhs) { return lhs.val > rhs.val; }
-    constexpr friend bool operator>=(Color lhs, Color rhs) { return lhs.val >= rhs.val; }
-
-    inline friend std::ostream& operator<<(std::ostream& os, Color rhs) { return os << rhs.val; }
+    inline friend std::ostream& operator<<(std::ostream& os, LegacyARGBU8Color rhs) { return os << uint32_t(rhs); }
+    inline friend std::istream& operator>>(std::istream& is, LegacyARGBU8Color& rhs) {
+        uint32_t val{};
+        is >> val;
+        rhs = val;
+        return is;
+    }
 
-    uint32_t val{};
+    // Legacy ordering (argb, not rgba):
```

Comment:
Mhh, I actually added this comment exactly to the place where someone will get curios and wants to start a refactoring.
I can add it above, but I don't think it will be a good idea to remove that line there.

Reply:
True. I rather meant copy rather than `std::move` :wink:
It's more that this is exactly the context one would need to understand the naming of the Sruct (is legacy meaning we have a newer/better colour struct that one should use, or that the ARGB ordering is legacy).

Reply:
Never mind, I chose to change the internal representation. Also, I removed the union, this may prevent implementation defined behavior. This will slightly increase the overhead due to byte swapping. But this will only have an effect on serialization.